lib/codegen/meta-python/base/instructions.py

```diff
@@ -144,6 +145,34 @@
         """,
         ins=(x, JT), is_branch=True)
 
+jump_table_entry = Instruction(
+    'jump_table_entry', r"""
+    Get an entry from a jump table.
+
+    Load an address from a specific entry in a jump table.
+    """,
+    ins=(x, JT), outs=addr)
+
+add_jump_table_base = Instruction(
+    'add_jump_table_base', r"""
+    Add the base address of a jump table to a value.
+
+    This is used for jump tables wherein the entries are stored relative to
+    the base of jump table. In order to use these, generated code should first
+    load an entry using ``jump_table_entry``, then use this instruction to add
+    the relative base back to it.
+    """,
+    ins=(addr, JT), outs=addr)
```

Comment:
I wonder if we could name this something like `decode_jump_table_entry`? It'd still be an add in the code as we know it today, but I think this name would help express that jump tables could be encoded in other ways too.

Reply:
Yeah this makes sense. In the future we’d  need to have this expand to different instructions depending on the jump table implementation? How would you want that to work? Wouldn’t we still need a specific `add_jump_table_base` instruction, or is there some way around that?

Reply:
That's a good question, and thinking about it makes me think that I may have been overthinking this ;-). Let's stick with `add_jump_table_base` for now.

It helps that this instruction isn't something that frontends should be producing directly, so it won't be too painful if we need to make changes.